core/src/test/java/org/apache/iceberg/TestSchemaID.java

```diff
@@ -0,0 +1,164 @@
+    );
+
+    validateTwoSchemas(updatedSchema, oldSchema);
+    validateSnapshotsAndHistoryEntries(1);
+
+    // remove file from table
+    table.newDelete().deleteFile(FILE_A).commit();
+
+    validateTwoSchemas(updatedSchema, oldSchema);
+
+    List<Snapshot> snapshots = Lists.newArrayList(table.snapshots());
+    Assert.assertEquals("Number of snapshot should match",
+        2, snapshots.size());
+    Assert.assertEquals("First schema id within snapshots should match",
+        Integer.valueOf(0), snapshots.get(0).schemaId());
+    Assert.assertEquals("Second schema id within snapshots should match",
+        Integer.valueOf(1), snapshots.get(1).schemaId());
+
+    Assert.assertEquals("Number of history entries should match",
+        2, table.history().size());
+    Assert.assertEquals("First history entry id within snapshots should match",
+        Integer.valueOf(0), table.history().get(0).schemaId());
+    Assert.assertEquals("Second history entry id within snapshots should match",
+        Integer.valueOf(1), table.history().get(1).schemaId());
+
+    // add files to table
+    table.newAppend().appendFile(FILE_A2).commit();
+
+    validateTwoSchemas(updatedSchema, oldSchema);
+
+    snapshots = Lists.newArrayList(table.snapshots());
+    Assert.assertEquals("Number of snapshot should match",
+        3, snapshots.size());
+    Assert.assertEquals("First schema id within snapshots should match",
+        Integer.valueOf(0), snapshots.get(0).schemaId());
+    Assert.assertEquals("Second schema id within snapshots should match",
+        Integer.valueOf(1), snapshots.get(1).schemaId());
+    Assert.assertEquals("Third schema id within snapshots should match",
+        Integer.valueOf(1), snapshots.get(2).schemaId());
+
+    Assert.assertEquals("Number of history entries should match",
+        3, table.history().size());
+    Assert.assertEquals("First history entry id within snapshots should match",
+        Integer.valueOf(0), table.history().get(0).schemaId());
+    Assert.assertEquals("Second history entry id within snapshots should match",
+        Integer.valueOf(1), table.history().get(1).schemaId());
+    Assert.assertEquals("Third history entry id within snapshots should match",
+        Integer.valueOf(1), table.history().get(2).schemaId());
+  }
+
+  private void validateSingleSchema() {
+    Assert.assertEquals("Current schema ID should match",
+        0, table.schema().schemaId());
+    assertSameSchemaMap(ImmutableMap.of(0, table.schema()), table.schemas());
+  }
+
+  private void validateTwoSchemas(Schema updatedSchema, Schema oldSchema) {
+    Assert.assertEquals("Current schema ID should match",
+        1, table.schema().schemaId());
+    Assert.assertEquals("Current schema should match",
+        updatedSchema.asStruct(), table.schema().asStruct());
+    assertSameSchemaMap(ImmutableMap.of(0, oldSchema, 1, updatedSchema), table.schemas());
+  }
+
+  private void validateSnapshotsAndHistoryEntries(int numElement) {
```

Comment:
You could have this helper method take a `List<Integer>` containing the schema ids instead of just an int (number of entries). Then `testSchemaIdChangeInSchemaUpdate` would be able to use this helper method for the cases after updating the schema. `snapshots.size()` should then match the `List`'s size, and the `snapshot.schemaId()` in each snapshot in `snapshots` should match the corresponding `Integer` in the `List`.

Reply:
Great suggestion, this makes the code in this class much easier to read; thank you!